hyper/common/exceptions.py

```diff
@@ -71,3 +71,10 @@ class MissingCertFile(Exception):
     The certificate file could not be found.
     """
     pass
+
+
+class ProxyError(Exception):
```

Comment:
I feel that it should extend a more generic `ConnectionError` exception, but there's no such common exception. But there's one in `hyper.http20.exceptions.ConnectionError`. Maybe we could introduce it here?